src/main/java/com/bookappstore/repository/user/UserRepository.java

```diff
@@ -3,7 +3,12 @@
 import com.bookappstore.model.User;
 import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("FROM User u LEFT JOIN FETCH u.roles r "
+            + "WHERE u.email = :email "
+            + "AND u.isDeleted = FALSE "
+            + "AND r.isDeleted = FALSE")
```

Comment:
What is the reason for using `@Query` here? If for JOIN FETCH only - better use `@EntityGraph`

Reply:
Nice tip, thanks